python/huggingfaceserver/huggingfaceserver/vllm_completions.py

```diff
@@ -0,0 +1,452 @@
+        top_p=request.top_p,
+        seed=request.seed,
+        stop=request.stop,
+        logprobs=request.logprobs,
+        max_tokens=request.max_tokens if not echo_without_generation else 1,
+        logits_processors=logits_processors,
+        prompt_logprobs=request.logprobs if request.echo else None,
+    )
+
+
+class OpenAIServingCompletion:
+
+    def __init__(self, engine: AsyncLLMEngine, served_model: str):
+        self.engine = engine
+        self.served_model = served_model
+
+        self.max_model_len = 0
+        self.tokenizer = None
+
+        try:
+            event_loop = asyncio.get_running_loop()
+        except RuntimeError:
+            event_loop = None
+
+        if (
+            event_loop is not None and event_loop.is_running()
+        ):  # If the current is instanced by Ray Serve, there is already a running event loop
+            event_loop.create_task(self._post_init())
+        else:  # When using single vLLM without engine_use_ray
+            loop = asyncio.new_event_loop()
+            loop.run_until_complete(self._post_init())
+            loop.close()
+
+    async def create_completion(self, request: CreateCompletionRequest):
+        """Completion API similar to OpenAI's API.
+
+        See https://platform.openai.com/docs/api-reference/completions/create
+        for the API specification. This API mimics the OpenAI Completion API.
+
+        NOTE: Currently we do not support the following feature:
+            - suffix (the language models we currently support do not support
+            suffix)
+        """
+        error_check_ret = await self._check_model(request)
+        if error_check_ret is not None:
+            return error_check_ret
+
+        # Return error for unsupported features.
+        if request.suffix is not None:
+            return self.create_error_response("suffix is not currently supported")
+
+        model_name = request.model
+        request_id = f"cmpl-{random_uuid()}"
+        created_time = int(time.time())
+
+        # Schedule the request and get the result generator.
+        generators = []
+        try:
+            sampling_params = to_sampling_params(request)
+            prompt_is_tokens, prompts = parse_prompt_format(request.prompt)
+
+            for i, prompt in enumerate(prompts):
+                if prompt_is_tokens:
+                    input_ids = self._validate_prompt_and_tokenize(
+                        request, prompt_ids=prompt
+                    )
+                else:
+                    input_ids = self._validate_prompt_and_tokenize(
+                        request, prompt=prompt
+                    )
+
+                generators.append(
+                    self.engine.generate(
+                        prompt,
+                        sampling_params,
+                        f"{request_id}-{i}",
+                        prompt_token_ids=input_ids,
+                    )
+                )
+        except ValueError as e:
```

Comment:
KServe should be able to handle the error response if you raise the `InvalidInput` exception
https://github.com/kserve/kserve/blob/9fd262f408ce5939d3d6e179fa789d5a34c256f9/python/kserve/kserve/protocol/rest/server.py#L213

Reply:
We can add ValueError to the above list